From 7ca519a5e269ecfe51c5fea6cd076b7c77536b2b Mon Sep 17 00:00:00 2001
From: Michael Simacek <msimacek@redhat.com>
Date: Mon, 23 Nov 2015 12:52:21 +0100
Subject: [PATCH] Port to current maven-dependency-tree

---
 .../updatecenter/mavenplugin/SonarPluginMojo.java  | 23 ++++++++++++----------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/sonar-packaging-maven-plugin/src/main/java/org/sonar/updatecenter/mavenplugin/SonarPluginMojo.java b/sonar-packaging-maven-plugin/src/main/java/org/sonar/updatecenter/mavenplugin/SonarPluginMojo.java
index b4570fc..e9d8b7b 100644
--- a/sonar-packaging-maven-plugin/src/main/java/org/sonar/updatecenter/mavenplugin/SonarPluginMojo.java
+++ b/sonar-packaging-maven-plugin/src/main/java/org/sonar/updatecenter/mavenplugin/SonarPluginMojo.java
@@ -38,9 +38,10 @@ import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
 import org.apache.maven.plugins.annotations.ResolutionScope;
-import org.apache.maven.shared.dependency.tree.DependencyNode;
-import org.apache.maven.shared.dependency.tree.DependencyTreeBuilder;
-import org.apache.maven.shared.dependency.tree.DependencyTreeBuilderException;
+import org.apache.maven.project.DefaultProjectBuildingRequest;
+import org.apache.maven.shared.dependency.graph.DependencyGraphBuilder;
+import org.apache.maven.shared.dependency.graph.DependencyGraphBuilderException;
+import org.apache.maven.shared.dependency.graph.DependencyNode;
 import org.codehaus.plexus.archiver.jar.JarArchiver;
 import org.codehaus.plexus.util.FileUtils;
 import org.sonar.updatecenter.common.FormatUtils;
@@ -93,7 +94,7 @@ public class SonarPluginMojo extends AbstractSonarPluginMojo {
   private MavenArchiveConfiguration archive = new MavenArchiveConfiguration();
 
   @Component
-  private DependencyTreeBuilder dependencyTreeBuilder;
+    private DependencyGraphBuilder dependencyTreeBuilder;
 
   /**
    * The artifact repository to use.
@@ -300,7 +301,7 @@ public class SonarPluginMojo extends AbstractSonarPluginMojo {
     return null;
   }
 
-  private List<String> copyDependencies() throws IOException, DependencyTreeBuilderException {
+    private List<String> copyDependencies() throws IOException, DependencyGraphBuilderException {
     List<String> ids = new ArrayList<String>();
     List<String> libs = new ArrayList<String>();
     File libDirectory = new File(getAppDirectory(), LIB_DIR);
@@ -326,7 +327,7 @@ public class SonarPluginMojo extends AbstractSonarPluginMojo {
     return artifact.getArtifactId() + "-" + artifact.getVersion() + "." + artifact.getArtifactHandler().getExtension();
   }
 
-  private Set<Artifact> getNotProvidedDependencies() throws DependencyTreeBuilderException {
+    private Set<Artifact> getNotProvidedDependencies() throws DependencyGraphBuilderException {
     Set<Artifact> result = new HashSet<Artifact>();
     Set<Artifact> providedArtifacts = getSonarProvidedArtifacts();
     for (Artifact artifact : getIncludedArtifacts()) {
@@ -363,13 +364,15 @@ public class SonarPluginMojo extends AbstractSonarPluginMojo {
     return false;
   }
 
-  private Set<Artifact> getSonarProvidedArtifacts() throws DependencyTreeBuilderException {
+    private Set<Artifact> getSonarProvidedArtifacts() throws DependencyGraphBuilderException {
     Set<Artifact> result = new HashSet<Artifact>();
     ArtifactFilter artifactFilter = new ScopeArtifactFilter(Artifact.SCOPE_RUNTIME);
     // We need to use Maven 2 dependency tree in order to get omitted dependencies
-    DependencyNode rootNode = dependencyTreeBuilder.buildDependencyTree(getProject(),
-      localRepository,
-      artifactFilter);
+        DefaultProjectBuildingRequest request = new DefaultProjectBuildingRequest();
+        request.setProject(getProject());
+        request.setRepositorySession(getSession().getRepositorySession());
+        DependencyNode rootNode = dependencyTreeBuilder.buildDependencyGraph(request,
+                artifactFilter);
     searchForSonarProvidedArtifacts(rootNode, result, false);
     return result;
   }
-- 
2.5.0

